_pages/home/release-notes/checkout-libraries-release-notes.md

```diff
@@ -0,0 +1,378 @@
+
+```typescript
+return this.checkoutDeliveryAddressConnector
+  .createAddress(userId, cartId, payload)
+  .pipe(
+    tap(() => {
+      ...
+    }),
+    map((address) => {
+      ...
+    }),
+    tap((address) =>
+      this.eventService.dispatch( // <-- Event dispatching
+        {
+          userId,
+          cartId,
+          address,
+        },
+        CheckoutDeliveryAddressCreatedEvent
+      )
+    )
+```
+
+Event listeners are flexible as we can inject any number of dependencies to perform certain actions. Moreover, in this case, it fires additional events, such as [`CheckoutResetDeliveryModesEvent`](https://sap.github.io/spartacus-docs/event-service/#list-of-spartacus-events) and [`CheckoutResetQueryEvent`](https://sap.github.io/spartacus-docs/event-service/#list-of-spartacus-events).
+
+```typescript
+  protected onDeliveryAddressChange(): void {
+    this.subscriptions.add(
+      this.eventService
+        .get(CheckoutDeliveryAddressCreatedEvent)
+        .subscribe(({ userId, cartId }) => {
+          this.globalMessageService.add(
+            { key: 'addressForm.userAddressAddSuccess' },
+            GlobalMessageType.MSG_TYPE_CONFIRMATION
+          );
+
+          this.eventService.dispatch(
+            { userId, cartId },
+            CheckoutResetDeliveryModesEvent
+          );
+
+          this.eventService.dispatch({}, CheckoutResetQueryEvent);
+        })
+    );
+    ...
+```
+
+The example for [`CheckoutResetQueryEvent`](https://sap.github.io/spartacus-docs/event-service/#list-of-spartacus-events) being fired is to reset our checkout details query from `CheckoutQueryFacade` due to the nature of our commands and queries, where we can [reload or reset a query](https://sap.github.io/spartacus-docs/commands-and-queries/#queries-overview).
+
+### How to properly make use of queries in checkout
+
+Example: [CheckoutQueryFacade](https://github.com/SAP/spartacus/blob/7bba93e0e6b75024371361b169348cfaebeb2c7b/feature-libs/checkout/base/core/facade/checkout-query.service.ts) and [CheckoutDeliveryAddressComponent](https://github.com/SAP/spartacus/blob/7bba93e0e6b75024371361b169348cfaebeb2c7b/feature-libs/checkout/base/components/checkout-delivery-address/checkout-delivery-address.component.ts)
+
+The 'checkout details query' that is injected in other checkout 'facade' services is used to get the current checkout details, which contains information on `delivery address, delivery mode, and payment information` when using base checkout. When using b2b checkout, you will have additional data like `purchase order number`, `cost centers`, and `payment type`.
+
```

Comment:
I would like to add a "lead-in" sentence here that says what this code snippet is, such as "The following is an example of the checkout details query".

Is it in fact an example of a checkout details query? If not, how can we refer to this example? What are we showing here? 🙂

Reply:
I think that lead-in makes sense

We're trying to show a checkout example on how-to use queries in checkout.

So we're saying: here is the 'facade' and a 'component' in our spartacus source repo. You should follow the 'flow' on how these are intertwined to make use of commands and query (specifically query)